activerecord/lib/active_record/autosave_association.rb

```diff
@@ -176,7 +176,8 @@ def define_non_cyclic_method(name, &block)
         # check if the save or validation methods have already been defined
         # before actually defining them.
         def add_autosave_association_callbacks(reflection)
-          save_method = :"autosave_associated_records_for_#{reflection.name}"
+          method_suffix = reflection.options[:class_name].try(:underscore)
```

Comment:
if you are using `try` I assume that means you expect `nil` as a possibility, in which case `save_method` will end up with `_` as the last character. You may want to build it as two-part array compacted and joined by `_`, or conditionally append method suffix only if it's not nil.